lib/contracts/genesisscheme.js

```diff
@@ -0,0 +1,186 @@
+"use strict";
+import { Utils } from "../utils";
+const Avatar = Utils.requireContract("Avatar");
+import { AbsoluteVote } from "./absoluteVote.js";
+
+const dopts = require("default-options");
+import { getDeployedContracts } from "../contracts.js";
+import { ExtendTruffleContract, ArcTransactionResult } from "../ExtendTruffleContract";
+
+const SolidityContract = Utils.requireContract("GenesisScheme");
+
+export class GenesisScheme extends ExtendTruffleContract(SolidityContract) {
+  static async new() {
+    const contract = await SolidityContract.new();
+    return new this(contract);
+  }
+
+  /**
+   * Create a new DAO
+   */
+  async forgeOrg(opts = {}) {
+    /**
+       * See these properties in DaoConfig
+     */
+    const defaults = {
+      name: undefined,
+      tokenName: undefined,
+      tokenSymbol: undefined,
+      founders: [],
+      universalController: true
+    };
+
+    const options = dopts(opts, defaults, { allowUnknown: true });
+
+    if (!options.name) {
```

Comment:
this works with undefined? I thought you had to check typeof === "undefined"?

Reply:
dopts handles the undefined check, but it has a weakness that it considers "null" to be a valid value of a required property (because null isn't undefined).  So we do a manual check.

In any case, this would also catch undefined.